cost-analyzer/values.yaml

```diff
@@ -153,6 +153,8 @@ saml:
   #metadataSecretName: "kubecost-authzero-metadata" # One of metadataSecretName or idpMetadataURL must be set. defaults to metadataURL if set
   idpMetadataURL: "https://dev-elu2z98r.auth0.com/samlp/metadata/c6nY4M37rBP0qSO1IYIqBPPyIPxLS8v2"
   appRootURL: "http://localhost:9090" # sample URL
+  authTimeout: 1440 # number of minutes the JWT will be valid
```

Comment:
How is this being passed into the app? Does there need to be an environment variable or something?

Reply:
There's an [existing configmap](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/cost-analyzer/templates/cost-analyzer-saml-config-map-template.yaml) for SAML which is [already being mounted](https://github.com/kubecost/cost-analyzer-helm-chart/blob/4a6b07dfb1679983b6dc655e8aac2fb74b735cec/cost-analyzer/templates/cost-analyzer-deployment-template.yaml#L168-L172). 

Actually, I see now it doesn't actually get mounted unless rbac is enabled... is there a reason for this? Because this configmap contains things not related to rbac as well (not just this PR's changes). I think we should probably change this to always mount if Values.saml.enabled.

Reply:
I'd avoid making such a foundational change and just  also pass/moun` these as env variables to not blow up the scope here. 

Once you do that, I'd also double check that authTimeout gets parsed correctly as an int when coming through an env variable since it won't be json serialized.

Reply:
Is there a particular reason why we view changing this configmap mount condition as foundational? AFAICT this configmap is always created if SAML is enabled.

Otherwise I feel strongly like creating two more env variables to pass values that are already in a configmap redundant. Increasing our number of env variables to avoid mounting a configmap that we both already try to unmarshal if SAML is enabled AND mount anyway if RBAC is enabled seems incorrect.

Reply:
It's beyond the scope of the PR to make that change, it's a significant change in functionality that may regress users since we'd be introducing a configmap and removing env variables. I would make the minimal change to unblock the feature then file a follow up issue if you feel strongly.

Reply:
I'm a bit confused by the current state of the world, here. I'm going to try to summarize what I see here and ask some questions.

From doing a little bit of spelunking, it looks to me like we have [a few env vars related to SAML](https://github.com/kubecost/cost-analyzer-helm-chart/blob/4a6b07dfb1679983b6dc655e8aac2fb74b735cec/cost-analyzer/templates/cost-analyzer-deployment-template.yaml#L597-L622):
- `SAML_ENABLED` (makes sense, this turns the functionality on)

And then we map some values under the `saml:` key:
- `IDP_URL`
- `SP_HOST`
- `GLUU_SAML_PROVIDER`
- `NAME_ID_FORMAT`
- and `SAML_RBAC_ENABLED` 

But we _also_ turn ALL of the values under `saml:` into the ConfigMap Kaelan is talking about. That's done [like this](https://github.com/kubecost/cost-analyzer-helm-chart/blob/4a6b07dfb1679983b6dc655e8aac2fb74b735cec/cost-analyzer/templates/cost-analyzer-saml-config-map-template.yaml#L11): 
```yaml
saml.json: '{{ toJson .Values.saml }}'
```

That ConfigMap is only mounted if `.Values.saml.rbac.enabled=true`, as Kaelan has noted.

@AjayTripathy Here are some questions that I hope will bring some clarity:
1. Why do we serialize everything under `saml:` into the ConfigMap?
2. Why do we pass only some `saml:` values as env vars?
3. Why do we only mount the ConfigMap if `.Values.saml.rbac.enabled=true`?
4. Are there values under `saml:` that should always be in the ConfigMap (instead of env var)?
5. Are there values under `saml:` that should always be in an env var (instead of the ConfigMap)?




I think there's definitely a ticket worth opening here to standardize how SAML config gets into the application. I think our discussion here can inform some documentation and, eventually, that ticket.

---

Here's a more direct contribution to this conversation, based on my current understanding:

A brief tour into the code suggests that mounting the ConfigMap but keeping `.Values.saml.rbac.enabled=false` shouldn't have adverse effects, see this snippet for example (other code paths are similar):
https://github.com/kubecost/kubecost-cost-model/blob/2f61e609cfae098886be61208831ed3439b66187/pkg/cmd/costmodel/auth.go#L204

I don't think Kaelan is suggesting that we remove any environment variables.

I think that he has read the code, found that much of SAML config is being pulled from the ConfigMap, and decided to follow what he found and implement the new configuration in the ConfigMap instead of environment variables.

I don't think he's suggesting we introduce a new ConfigMap, either; he'd like to always _mount_ the already-existing ConfigMap to have access to these new config values. I think this is the crux: I don't quite understand why this isn't allowed. I hope answers to the above questions will get us all on the same page.

Reply:
@michaelmdresser @AjayTripathy Apologies for the long discussion. After looking at the RBAC code, I think there is a possibility that unintended effects could occur if the configmap is mounted while RBAC is disabled. While I would like to address these in the interests of clarity, I understand that changes may impact RBAC functionality and so will leave that out of this PR.

To that end, I have implemented and tested the proposed env variables. If I could get another look, that would be great.